com.github.Inochi2D.Inochi-Creator.yml

```diff
@@ -0,0 +1,167 @@
+    - "--device=all" #OpenGL rendering, webcams (may be required? not sure)
+    - "--socket=x11"
+    - "--share=ipc" # The X and Wayland sockets are permanently exposed, because kdialog takes longer to start if it only has X access, and doesn't work at all if only using Wayland.
+    - "--socket=wayland"
+    - "--filesystem=host" #This is a dev tool, it should probably have as much access to the host as it can. It also doesn't yet utilize portals.
+    - "--env=XDG_CURRENT_DESKTOP=KDE" #Hopefully fixes the issues it has with running on non-KDE desktops.
+modules:
+  # --- Tools ---
+  - name: xprop
+    sources: 
+      - type: archive
+        url: https://xorg.freedesktop.org/archive/individual/app/xprop-1.2.4.tar.bz2
+        sha256: "8c77fb096e46c60032b7e2bde9492c3ffcc18734f50b395085a5f10bfd3cf753"
+
+  - name: zenity
+    buildsystem: meson
+    sources:
+      - type: archive
+        url: https://download.gnome.org/sources/zenity/3.41/zenity-3.41.0.tar.xz
+        sha256: 19b676c3510e22badfcc3204062d432ba537402f5e0ae26128c0d90c954037e1
+
+  - name: kdialog
+    buildsystem: cmake
+    sources:
+      - type: git
+        url: https://github.com/KDE/kdialog.git
+        tag: v22.08.1
+
+
+  # --- Inochi Creator ---
+  - name: inochi-creator
+    buildsystem: simple
+    build-options:
+      append-path: /usr/lib/sdk/ldc/bin
+    build-commands:
+      - dub add-local ./deps/bindbc-imgui/ 0.7.0+build.20.g237a209;
+        dub add-local ./deps/facetrack-d/ 0.6.2+build.32.g9e1b2b1;
+        dub add-local ./deps/fghj/ 1.0.0;
+        dub add-local ./deps/gitver/ 1.5.0;
+        dub add-local ./deps/i18n/ 1.0.1+build.1.ge4a7f0b;
+        dub add-local ./deps/inmath/ 1.0.3+build.4.gec62993;
+        dub add-local ./deps/inochi2d/ 0.7.2+build.17.g0e32b36;
+        dub add-local ./deps/psd-d/ 0.6.2;
+        dub add-local ./deps/vmc-d/ 1.1.1+build.4.gb32fb96;
+        dub add-local ./deps/bindbc-loader/ 1.0.1;
+        dub add-local ./deps/bindbc-opengl/ 1.0.2;
+        dub add-local ./deps/bindbc-sdl/ 1.1.3;
+        dub add-local ./deps/imagefmt/ 2.1.2;
+        dub add-local ./deps/mir-algorithm/ 3.14.11;
+        dub add-local ./deps/mir-core/ 1.1.109;
+        dub add-local ./deps/semver/ 0.3.4;
+        dub add-local ./deps/silly/ 1.1.1;
+        dub add-local ./deps/tinyfiledialogs/ 0.10.1;
+        dub build --skip-registry=all --compiler=ldc2 --build=release --config=full;
+      - install -m 755 ./out/inochi-creator /app/bin/inochi-creator
+
+
+      - install -Dm644 ./inochi-creator.appdata.xml -t /app/share/metainfo/com.github.Inochi2D.Inochi-Creator.metainfo.xml
+      - install -Dm644 ./res/logo.png /app/share/icons/hicolor/128x128/apps/com.github.Inochi2D.Inochi-Creator.png
+      - install -Dm644 ./res/logo_256.png /app/share/icons/hicolor/256x256/apps/com.github.Inochi2D.Inochi-Creator.png
+      - install -Dm644 ./inochi-creator.desktop -t /app/share/applications/com.github.Inochi2D.Inochi-Creator.desktop
+
+    sources:

```

Comment:
```
install: cannot stat 'inochi-creator.appdata.xml': No such file or directory
```

because the file isn't listed here.

But then they are in the `res` directory of the source, so we shouldn't even have them in the repo.

Reply:
Ah, yeah, you're right. Mb. Wouldn't they need to be modified for Flatpak, though? For example, the original `appdata.xml` file uses the ID `inochi-creator`, and not `com.github.Inochi2D.Inochi-Creator`

Reply:
1. you can patch
2. you should fix it upstream. Their file is invalid.
3. I'm prretty sure that if you install it with the original and use `rename-appdate-file` will deal with it.

None of our requirements are specific to Flathub, they are just more strict.

Reply:
Hm, I'll give the `rename-*-file` flags a try locally and push if they work. The original plan was to get this submitted, and then push the modified files to upstream.